### ISSUE TITLE
Add search support for datasetName and datasetID

### ISSUE THREAD

**Issue by MortenHofft** [gbif/pipelines#662]
Prompted by https://github.com/gbif/portal-feedback/issues/3006 and https://github.com/gbif/portal-feedback/issues/3026

The desire in above issues it to have a way to search for occurrences within or across datasets using `datasetName` https://dwc.tdwg.org/terms/#dwc:datasetName and `datasetID` https://dwc.tdwg.org/terms/#dwc:datasetID. 

And to allow multiple values when the publisher delimits them by `|`.

So if a publisher provides an occurrence with`datasetName: projectA|projectB` then it should be possible in the API to find that occurrence by adding a filter like `datasetName=projectA` (a keyword filter in ES terminology I guess?)

I do not believe we need a suggest endpoint for this.

**Comment by MortenHofft** [comment]
I just realise that I haven't address these questions:

**Current data that use pipes in a different way**
The documentation currently do not say anything about using a pipe separator to split multiple values for `datasetName`.
So we have data that use `|`, but where it probably isn't intended as 2 names as such but as a hierarchy. E.g. `Natural History Museum Denmark | Herpetology Collection`
https://api.gbif.org/v1/occurrence/search?occurrenceID=5d867879-6e7d-4463-8773-e005a6a9731d

Would it make sense (and feasible) to evaluate how much data would be misinterpreted this way and try to get it changed at source?

**Current response is a string**
This issue suggest that we interpret this an array of keywords for search. Should the response format stay the same as now: a string. Or should it be changed to an array of strings? And where do we draw the line for API stability?

**Comment by rukayaj** [comment]
Anything I can do to help with this?

**Comment by MortenHofft** [comment]
It is in UAT https://api.gbif-uat.org/v1/occurrence/search?datasetName=ebcc%20atlas%20of%20european%20breeding%20birds, but part of a larger feature release that influence all UIs and I've been slow at adapting to all the small changes.

**Comment by marcos-lg** [comment]
Deployed to PROD.

**Comment by rukayaj** [comment]
Thanks for this! I was just wondering if it will get added to the front end gui search options https://www.gbif.org/occurrence/search any time soon?

**Comment by MortenHofft** [comment]
That wasn't the plan to add those in the UI (gbif.org). They seem to be very internal and mainly intended for publisher API users. At least that is how we have discussed it so far.

**Comment by rukayaj** [comment]
We have a user who wants to be able to download all occurrences with a particular datasetName, and then have a DOI to cite in a paper. The records are a small subset of records within this huge dataset https://www.gbif.org/dataset/b124e1e0-4755-430f-9eab-894f25a9b59c, datasetName = NØF-vannfugltellinger

Works fine on the API: [https://api.gbif.org/v1/occurrence/search?datasetname=NØF-vannfugltellinger](https://api.gbif.org/v1/occurrence/search?datasetname=N%C3%98F-vannfugltellinger)

**Comment by rondlg** [comment]
Hi, we also have users who want to be able to search on that term. In fact their grouo was one of the primary reasons why we sent in the request. That said the api access is very welcome.

**Comment by rukayaj** [comment]
I've had another request for the GBIF UI to allow searching by datasetName. Use case: 

Colleagues have received funding from a particular funding agency and have published samples associated with these projects in our Collections Management System which are grouped using datasetName. It would then be useful to be able to search out the relevant data in GBIF.
 
The same data field in the CMS is also used to identify other datasets, such as e.g. all accessions used for the Arctic collections paper we published last year (which I guess was when we started talking about datasetName).
 
While it’s great that it’s available via the API, this is not a solution that is easily accessible to many users, and hence it would be very nice to be able to search for it in the standard GBIF UI.
